cmd/thanos/receive.go

```diff
@@ -311,6 +311,8 @@ func runReceive(
 			httpserver.WithGracePeriod(time.Duration(*conf.httpGracePeriod)),
 			httpserver.WithTLSConfig(*conf.httpTLSConfig),
 		)
+		var lastDownscalePrepareTimestamp *int64 = nil
```

Comment:
this implementation assumes the pod won't restart, to preserve the state, we might need to persist it on local PVC too, any thoughts?

Reply:
yes but it's safe to lose this state. rollout operator will only do downscale if the returned timestamp is more than 30 seconds ago. If we lose this state, downscale will never accidentally happen and, in the worst case, a scheduled downscale will be delayed for 30 seconds.